apps/api/src/lib/sort.ts

```diff
@@ -0,0 +1,111 @@
+import { ZodEffects, ZodOptional, ZodString, z } from "zod";
+
+/**
+ * Build a Zod schema that accepts
+ *   "field" | "-field"
+ * for any field name in the supplied list.
+ * @param fields - The list of fields to accept
+ * @returns A Zod schema that accepts "field" | "-field" for any field name in the supplied list
+ */
+export function makeSortFieldSchema<const FIELDS extends readonly string[]>(
```

Comment:
this basically just lets use pass a `["field"]`, and the result will transform it to `["field","-field"]` so that zod can validate the sort asc/desc.

there are _many_ ways to do this with zod, though.